### ISSUE TITLE
[python3] Build error

### ISSUE THREAD

**Issue by madlemur** [microsoft/vcpkg#31879]
### Host Environment

OS: Windows 11
Visual Studio: 2022
Python : 3.10

### To Reproduce

`cmake` (via VS 2022, vcpkg in Manifest mode)

### Failure logs
```
1> [CMake]   * boost-ptr-container[core]:x64-windows -> 1.82.0#2
1> [CMake]   * boost-python[core,python3]:x64-windows -> 1.82.0#2
1> [CMake]   * python3[core]:x64-windows -> 3.10.7#6
1> [CMake]   * zstd[core]:x64-windows -> 1.5.5
1> [CMake] Additional packages (*) will be modified to complete this operation.
1> [CMake] Restored 0 package(s) from C:\____\AppData\Local\vcpkg\archives in 1.25 ms. Use --debug to see more details.
1> [CMake] Installing 1/54 python3:x64-windows...
1> [CMake] Building python3[core]:x64-windows...
1> [CMake] -- Using cached python-cpython-v3.10.7.tar.gz.
1> [CMake] -- Cleaning sources at C:/opt/vcpkg/buildtrees/python3/src/v3.10.7-10c7c80a03.clean. Use --editable to skip cleaning for the packages you specify.
1> [CMake] -- Extracting source C:/opt/vcpkg/downloads/python-cpython-v3.10.7.tar.gz
1> [CMake] -- Applying patch 0001-only-build-required-projects.patch
1> [CMake] -- Applying patch 0003-use-vcpkg-zlib.patch
1> [CMake] -- Applying patch 0004-devendor-external-dependencies.patch
1> [CMake] -- Applying patch 0005-dont-copy-vcruntime.patch
1> [CMake] -- Applying patch 0008-python.pc.patch
1> [CMake] -- Applying patch 0009-bz2d.patch
1> [CMake] -- Applying patch 0010-dont-skip-rpath.patch
1> [CMake] -- Applying patch 0012-force-disable-curses.patch
1> [CMake] -- Applying patch 0013-configure-no-libcrypt.patch
1> [CMake] -- Applying patch 0014-fix-get-python-inc-output.patch
1> [CMake] -- Applying patch 0007-workaround-windows-11-sdk-rc-compiler-error.patch
1> [CMake] -- Using source at C:/opt/vcpkg/buildtrees/python3/src/v3.10.7-10c7c80a03.clean
1> [CMake] -- Building PCbuild/pcbuild.proj for Release
1> [CMake] CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
1> [CMake]     Command failed: "C:/Program Files (x86)/PowerShell/7/pwsh.exe" -noprofile -executionpolicy Bypass -nologo -file C:/opt/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary C:/opt/vcpkg/packages/python3_x64-windows/tools/python3/_freeze_importlib.exe -installedDir C:/opt/vcpkg/packages/python3_x64-windows/bin -verbose
1> [CMake]     Working Directory: C:/opt/vcpkg
1> [CMake]     Error code: **Access violation**
1> [CMake]     See logs for more information:
1> [CMake]       C:\opt\vcpkg\buildtrees\python3\copy-tool-dependencies-0-err.log
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   scripts/cmake/vcpkg_copy_tool_dependencies.cmake:9 (vcpkg_execute_required_process)
1> [CMake]   scripts/cmake/vcpkg_copy_tool_dependencies.cmake:41 (z_vcpkg_copy_tool_dependencies_search)
1> [CMake]   scripts/cmake/vcpkg_install_msbuild.cmake:98 (vcpkg_copy_tool_dependencies)
1> [CMake]   ports/python3/portfile.cmake:150 (vcpkg_install_msbuild)
1> [CMake]   scripts/ports.cmake:147 (include)
1> [CMake] error: building python3:x64-windows failed with: BUILD_FAILED
1> [CMake] Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
1> [CMake] Then check for known issues at:
1> [CMake]     https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+python3
1> [CMake] You can submit a new issue at:
1> [CMake]     https://github.com/microsoft/vcpkg/issues/new?title=[python3]+Build+error&body=Copy+issue+body+from+C%3A%2F____%2Fsource%2Frepos%2FHTTPServer%2Fout%2Fbuild%2Fx64-debug%2Fvcpkg_installed%2Fvcpkg%2Fissue_body.md
1> [CMake] 
1> [CMake] -- Running vcpkg install - failed
1> [CMake] CMake Error at C:/opt/vcpkg/scripts/buildsystems/vcpkg.cmake:893 (message):
1> [CMake]   vcpkg install failed.  See logs for more information:
1> [CMake]   C:\Users\kenne\source\repos\HTTPServer\out\build\x64-debug\vcpkg-manifest-install.log
1> [CMake] Call Stack (most recent call first):
1> [CMake]   C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.26/Modules/CMakeDetermineSystem.cmake:148 (include)
1> [CMake]   CMakeLists.txt:12 (project)
1> [CMake] -- Configuring incomplete, errors occurred!
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="cl.exe" -DCMAKE_CXX_COMPILER:STRING="cl.exe" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="C:/opt/vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:/____/source/repos/HTTPServer/out/install/x64-debug"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\____\source\repos\HTTPServer" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_C_COMPILER:STRING="cl.exe" -DCMAKE_CXX_COMPILER:STRING="cl.exe" -DCMAKE_TOOLCHAIN_FILE:FILEPATH="C:/opt/vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:/____/source/repos/HTTPServer/out/install/x64-debug"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Users\____\repos\HTTPServer" 2>&1"' returned with exit code: 1'.

```
[error-logs-x64-windows.txt](https://github.com/microsoft/vcpkg/files/11683061/error-logs-x64-windows.txt)
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/11683062/stdout-x64-windows.log)
[copy-tool-dependencies-0-err.log](https://github.com/microsoft/vcpkg/files/11683063/copy-tool-dependencies-0-err.log)

I can successfully build the same project for x64-linux via WSL. I have updated vcpkg. I have removed and reinstalled vcpkg.

**Comment by LilyWangLL** [comment]
Thanks for posting this issue. This is a bug of PowerShell 7.3.0 https://github.com/PowerShell/PowerShell/issues/18545, this has been fixed in PowerShell 7.3.1, please update your PowerShell and rebuild `python3`.

The related issue: https://github.com/microsoft/vcpkg/issues/28046

**Comment by madlemur** [comment]
Almost lost it, as I run PowerShell 7.3.4, but then remembered that I installed 7.3.4 _over_ 7.3.0, not as an upgrade. so I went in, uninstalled the old 7.3.0, and I appear to be in business. I'm hoping this is also the solution for other access violation issues I was having on the windows side of the build system elsewhere (it has the same smell).
